MUI_Core/Modules/DataText/Components/Quest.lua

```diff
@@ -0,0 +1,70 @@
+-- luacheck: ignore MayronUI self 143 631
+local tk, db, em, _, obj, L = _G.MayronUI:GetCoreComponents();
+local obj = _G.MayronObjects:GetFramework();
+local Quest = obj:CreateClass("Quest");
+
+-- CombatTimer Module ----------------
+
+MayronUI:Hook("DataTextModule", "OnInitialize", function(self)
+  self:RegisterComponentClass("quest", Quest);
+end);
+
+function Quest:__Construct(data, settings, dataTextModule)
+  data.settings = settings;
+
+  -- set public instance properties
+  self.TotalLabelsShown = 0;
+  self.HasLeftMenu = false;
+  self.HasRightMenu = false;
+  self.Button = dataTextModule:CreateDataTextButton();
+end
+
+function Quest:IsEnabled(data)
+  return data.enabled;
+end
+
+function Quest:SetEnabled(data, enabled)
+  data.enabled = enabled;
+
+  local listenerID = "DataText_Quest_OnChange";
+  if (enabled) then
+    if (not em:GetEventListenerByID(listenerID)) then
+      local listener = em:CreateEventListenerWithID(listenerID, function()
+        if (not self.Button) then return end
```

Comment:
Because you created self.Button inside the constructor then this should always be false? self.Button can only be a button. Not a big issue :)